src/v/cluster/controller_backend.cc

```diff
@@ -491,12 +491,39 @@ controller_backend::calculate_learner_initial_offset(
      * Initial learner start offset only makes sense for partitions with cloud
      * storage data
      */
+    if (auto tp_cfg = p->get_topic_config();
+        tp_cfg.has_value() && tp_cfg->get().is_internal()) {
+        vlog(clusterlog.trace, "{} is part of an internal topic", p->ntp());
```

Comment:
Wondering if this should be updated to just look at whether we're in the kafka namespace? In this branch, `archival_metadata_stm_factory::is_applicable_for()` doesn't consider the CO topic

For that matter, maybe it'd be more future-proof to make a static `archival_metadata_stm_factory::is_applicable_for(const model::ntp&)` and use that here and in the factory